pkg/controller/control.go

```diff
@@ -1051,17 +1051,28 @@ func (c *Controller) UnmapAt(length uint32, off int64) (int, error) {
 	c.RLock()
 
 	if off < 0 || off+int64(length) > c.size {
-		err := fmt.Errorf("EOF: Unmap of %v bytes at offset %v is beyond volume size %v", length, off, c.size)
+		// This is a legitimate error (which is handled by tgt/liblonghorn at a higher level). Since tgt/liblonghorn
+		// does not actually print the error, print it here.
+		err := fmt.Errorf("EOF: unmap of %v bytes at offset %v is beyond volume size %v", length, off, c.size)
+		logrus.WithError(err).Error("Failed to unmap")
 		c.RUnlock()
 		return 0, err
```

Comment:
NIT: return 0, io.ErrUnexpectedEOF

Reply:
I'm likely fine with this, but can you help clarify the goal here, as I am new to the conventions in this code. If we return `io.ErrUnexpectedEOF` here, it will end up in:

https://github.com/longhorn/longhorn-engine/blob/341436e7f3dac72449a30b3893b2fd5290785c07/pkg/dataconn/server.go#L110-L120

We will just treat it like any other error and push it back to liblonghorn with `msg.Type == TypeError` instead of `msg.Type == TypeEOF`. Is that the intention, or do we need to make additional changes?

Reply:
[`longhorn-engine/pkg/dataconn`](https://github.com/longhorn/longhorn-engine/blob/341436e7f3dac72449a30b3893b2fd5290785c07/pkg/dataconn/server.go#L110-L120) is the data transfer service between engine and replicas. The results are handled by engine like [this](https://github.com/longhorn/longhorn-engine/blob/f2418966602de5d3eb3ab7e7dfeb8dc38d214380/pkg/controller/replicator.go#L129). 
While the error in `longhorn-engine/pkg/controller/control` will be returned to the frontend iscsi layer.

This is a NIT suggestion anyway. I am fine with the existing implementation.